Symfony/CS/Fixer/Contrib/Php4ConstructorFixer.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+/*
+ * This file is part of the Symfony CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Matteo Beccati <matteo@beccati.com>
+ */
+class Php4ConstructorFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        foreach ($tokens->findGivenKind(T_CLASS) as $index => $token) {
+            // Is it inside a namespace?
+            $nspIndex = $tokens->getPrevTokenOfKind($index, array(array(T_NAMESPACE, 'namespace')));
+            if ($nspIndex) {
+                $nspName = $tokens->getNextMeaningfulToken($nspIndex); // Name
+                $nspBrace = $tokens->getNextMeaningfulToken($nspName); // { or ;
+
+                if ($tokens[$nspBrace]->equals(';')) {
+                    // Class is inside a (non-block) namespace
+                    continue;
+                } else {
+                    if ($index < $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $nspBrace)) {
+                        // Class is inside a block namespace
+                        continue;
```

Comment:
I don't understand why you continue the loop in these 2 cases. It looks to me that the real check is whether the ``$nspName`` token is actually a name, or a delimiter.